File: frontend/types/exam.ts
import type { Exam, Question, Result } from '@/types'
export type AnswerValue = string | string[]
export type ExamSession = { exam: Exam & { questions?: Question[] }; startAt: string; endAt: string; durationSeconds: number }
export type AttemptState = { examId: string; questionIndex: number; answers: Record<string, AnswerValue>; review: Record<string, boolean>; startAt: string; endAt: string; submitted: boolean; autosaving: boolean; lastSavedAt?: string }
export type ResultDetail = Result & { answers: Record<string, AnswerValue>; questions: Question[]; correct: number; totalQuestions: number }
export type ExamPayload = Pick<Exam, 'title' | 'subject' | 'duration'> & { description?: string; questions?: Question[] }
export type QuestionPayload = Omit<Question, 'id'>

File: src/main/java/com/examora/model/Exam.java
package com.examora.model;

import java.time.Instant;

public record Exam(
        String id,
        String title,
        String subject,
        String date,
        int duration,
        String status,
        int participants,
        Double averageScore,
        Instant startAt,
        Instant endAt
) {
    public Exam(String id, String title, String subject, String date, int duration, String status, int participants, Double averageScore) {
        this(id, title, subject, date, duration, status, participants, averageScore, null, null);
    }
}
